controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -20,40 +20,70 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/go-logr/logr"
 	v1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/api/v1alpha1"
 	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
 	"github.com/red-hat-storage/ocs-operator/controllers/storagecluster"
+	controllers "github.com/red-hat-storage/ocs-operator/controllers/storageconsumer"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
+
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	storageClassClaimFinalizer  = "storagesclassclaim.ocs.openshift.io"
+	StorageClassClaimAnnotation = "ocs.openshift.io.storagesclassclaim"
+)
+
 // StorageClassClaimReconciler reconciles a StorageClassClaim object
 // nolint
 type StorageClassClaimReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+	log    logr.Logger
+	ctx    context.Context
+
+	storageConsumer              *ocsv1alpha1.StorageConsumer
+	storageClassClaim            *ocsv1alpha1.StorageClassClaim
+	cephBlockPool                *rookCephv1.CephBlockPool
+	cephFilesystemSubVolumeGroup *rookCephv1.CephFilesystemSubVolumeGroup
+	cephClientRBDProvisioner     *rookCephv1.CephClient
+	cephClientRBDNode            *rookCephv1.CephClient
+	cephClientCephFSProvisioner  *rookCephv1.CephClient
+	cephClientCephFSNode         *rookCephv1.CephClient
+	cephResourcesByName          map[string]*ocsv1alpha1.CephResourcesSpec
+
+	namespace string
 }
 
 // +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephclients,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephfilesystemsubvolumegroups,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageconsumers/status,verbs=get;update;patch
 
 func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx, "StorageClassClaim", request)
-	ctx = ctrllog.IntoContext(ctx, log)
-	log.Info("Reconciling StorageClassClaim.")
+	r.log = ctrllog.FromContext(ctx, "StorageClassClaim", request)
+	r.ctx = ctrllog.IntoContext(ctx, r.log)
+	r.log.Info("Reconciling StorageClassClaim.")
 
 	// Fetch the StorageClassClaim instance
-	claim := &ocsv1alpha1.StorageClassClaim{}
-	if err := r.Client.Get(ctx, request.NamespacedName, claim); err != nil {
+	r.storageClassClaim = &ocsv1alpha1.StorageClassClaim{}
```

Comment:
I initialized the storageClassClaimm inside initReconcile

so you are suggesting to initialize storageCluster inside initReconcile method but it will require the name for the resources, right?